Descent3/menu.cpp

```diff
@@ -1398,6 +1455,76 @@ int DisplayLevelWarpDlg(int max_level) {
   hwnd.Destroy();
   return chosen_level;
 }
+
+// DisplayLevelSelectDlg
+// displays a list of levels associated with the selected mission
+int DisplayLevelSelectDlg(int max_level) {
```

Comment:
Probably you can add campaign name as second argument, so you don't have to create global variable `msnname`.